library/Fission/IPFS/Peer.hs

```diff
@@ -54,5 +61,43 @@ connect peer@(Peer peerID) = IPFSProc.run ["swarm", "connect"] (UTF8.textToLazyB
   (ExitFailure _ , _, _) -> Left $ CannotConnect peer
   (ExitSuccess   , _, _) -> Right ()
 
+peerAddressRe = mkRegex "^/ip[46]/([a-zA-Z0-9.:]*)/"
+
+extractIPfromPeerAddress peer = peer
+  & matchRegex peerAddressRe
+  <&> Data.List.head
+
+isExternalIPv4 :: Text -> Bool
+isExternalIPv4 ip = ip
+  & Text.unpack
+  & extractIPfromPeerAddress
+  & maybe "" Text.pack
+  & Net.IPv4.decode
+  <&> Net.IPv4.reserved
+  & maybe False not
+
+
+filterExternalPeers :: [Peer] -> [Peer]
+filterExternalPeers = Data.List.filter (isExternalIPv4 . peer)
```

Comment:
@expede Where would a good place to put util functions?

Reply:
If they're not going to be used elsewhere, you  can leave them in the module and not re-export them, or put it in a `where`. There's nothing wrong with having utility functions in the module itself, but if you wanted to split them out you can either use a generic `*.Util`, or something more semantic like `Fission.IPFS.Peer.External`.